### PR TITLE
Pushing BackdropFilter Mutator

### PR DESCRIPTION
**Pull request by JTKryptic**
Creates the container functions for the BackdropFilter mutation that will push the mutator to the mutator stack of each visited PlatformView. Continues from #34408 (Create blanket backdrop filter mutator) and to be combined with #34596 (PlatformView Blur for Backdrop Filter).

Related Issues:  [#43902](https://github.com/flutter/flutter/issues/43902)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by cyanglaz — CHANGES_REQUESTED** [review]

File: flow/embedded_views.h
Comment (JTKryptic) [reply flow/embedded_views.h:24]: Thank you!

File: flow/embedded_views.h
Comment (JTKryptic) [reply flow/embedded_views.h]: This threw an error when I called it so I left it as is, I will try it again now that the rest of the changes are made

File: flow/layers/backdrop_filter_layer.cc
Comment (JTKryptic) [reply flow/layers/backdrop_filter_layer.cc]: Done.

File: flow/embedded_views.h
Comment (JTKryptic) [reply flow/embedded_views.h]: Done.

**Comment by chinmaygarde** [comment]
@cyanglaz Looks like this is good for another review? @JTKryptic Can you address the presubmit failures please?

**Comment by cyanglaz** [comment]
@chinmaygarde 
@JTKryptic Is going to explore some other approach and I'm going to mark this PR as draft for now.

File: flow/layers/backdrop_filter_layer.cc
Comment (cyanglaz) [inline flow/layers/backdrop_filter_layer.cc:47]: Do we need this? If there are no platform views, `PushFilterToVisitedPlatformViews` is going to be a no-op anyway because the list is empty.

Comment (JTKryptic) [reply flow/layers/backdrop_filter_layer.cc:47]: It can be removed. I wasn't sure if it was better to check if we found a platform view first or to just run the method every time even if its no-op

Comment (cyanglaz) [reply flow/layers/backdrop_filter_layer.cc:47]: Yeah, it's ok to run the no-op every time. the check is redundant.

Comment (JTKryptic) [reply flow/layers/backdrop_filter_layer.cc]: done

File: flow/embedded_views.h
Comment (cyanglaz) [inline flow/embedded_views.h:244]: nits: PushImageFilter might be a better name.

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h
Comment (cyanglaz) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h:304]: nits:
```suggestion
  // A vector of visited platform view IDs.
```

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h
Comment (cyanglaz) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h:190]: complete the comment.

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h
Comment (cyanglaz) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h:187]: nits:
```suggestion
  // Pushes backdrop filter mutation to the mutator stack of each visited platform view.
```

File: flow/embedded_views.h
Comment (cyanglaz) [inline flow/embedded_views.h:370]: nits:
```suggestion
  // list of visited platform views.
```

File: flow/embedded_views.h
Comment (cyanglaz) [inline flow/embedded_views.h:374]: nits:
```suggestion
  // list of visited platform views.
```

**Review by cyanglaz — COMMENTED** [review]
Code LGTM % some nits. 

Waiting for unittest that tests the "push image filter to the mutator stack" logic before landing.

File: flow/embedded_views.h
Comment (cyanglaz) [inline flow/embedded_views.h:377]: nits:
Can add
// *See also: |PushVisitedPlatformView| for pushing platform view ids to the visited platform views list.

File: shell/common/shell_unittests.cc
Comment (cyanglaz) [inline shell/common/shell_unittests.cc:802]: The `queue` is not used, you can remove this

**Comment by cyanglaz** [comment]
@JTKryptic Could you also bring back the check list in the PR description?

**Review by cyanglaz — APPROVED** [review]
LGTM!
@LongCatIsLooong Do you mind doing a secondary review?

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm
Comment (cyanglaz) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm:327]: We should also add a test for this part.

**Review by LongCatIsLooong — APPROVED** [review]

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm:327]: I'm going to block this PR to land for now until [#34596](https://github.com/flutter/engine/pull/34596) is landed

**Review by cyanglaz — CHANGES_REQUESTED** [review]

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm:327]: Offline discussion. We will land this PR first and add the final test in a follow up PR

**Review by cyanglaz — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
This isn't compiling:
```
../../flutter/shell/common/shell_test_external_view_embedder.h:72:13: error: functions that differ only in their return type cannot be overloaded
  SkCanvas* CompositeEmbeddedView(int view_id) override;
  ~~~~~~~~~ ^
../../flutter/shell/common/shell_test_external_view_embedder.h:69:24: note: previous declaration is here
  EmbedderPaintContext CompositeEmbeddedView(int view_id) override;
  ~~~~~~~~~~~~~~~~~~~~ ^
1 error generated.
```

**Review by cyanglaz — COMMENTED** [review]
Still LGTM after resolving conflicts.

File: flow/layers/backdrop_filter_layer.cc
Comment (moffatman) [inline flow/layers/backdrop_filter_layer.cc:46]: The filter is being pushed but never cleared. The platform view gets an ever-increasing number of filters until app is unresponsive.

File: flow/layers/platform_view_layer.cc
Comment (moffatman) [inline flow/layers/platform_view_layer.cc:32]: Platform view is pushed but nothing ever clears that list, so it will have the same issue as the filters, it slows down rendering until app is unresponsive.

File: flow/layers/backdrop_filter_layer.cc
Comment (cyanglaz) [reply flow/layers/backdrop_filter_layer.cc:46]: Yes, we are fixing it here: https://github.com/flutter/engine/pull/34596

File: flow/layers/platform_view_layer.cc
Comment (cyanglaz) [reply flow/layers/platform_view_layer.cc:32]: Yes, we are fixing it here: https://github.com/flutter/engine/pull/34596

**Comment by cyanglaz** [comment]
Created an issue to track the platform view list not cleared bug: https://github.com/flutter/flutter/issues/109783

File: flow/layers/backdrop_filter_layer.cc
Comment (moffatman) [reply flow/layers/backdrop_filter_layer.cc:46]: I ran with a print statement of the number of mutators in vector_ in MutatorsStack::PushBackdropFilter, it was continuously increasing

**Comment by zanderso** [comment]
@cyanglaz In light of the P2 here https://github.com/flutter/flutter/issues/109831, this should be reverted, and relanded only with the fix.

**Pull request merged**
